### PR TITLE
proto(engine): remove error field from rpc response structure

### PR DESCRIPTION
**Pull request by gozssky**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/7180

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
None
```

### CODE REVIEW

File: engine/executor/server.go [hunk 1]
```diff
@@ -640,31 +641,20 @@ func (s *Server) keepHeartbeat(ctx context.Context) error {
...
-				case pb.ErrorCode_UnknownExecutor:
-					log.L().Info("heartbeat failed, will retry", zap.Error(err))
-					continue
-				case pb.ErrorCode_TombstoneExecutor:
-					return errors.ErrHeartbeat.GenWithStack("logic error: %s", resp.Err.GetMessage())
-				case pb.ErrorCode_MasterNotReady:
-					s.lastHearbeatTime = t
-					if rl.Allow() {
-						log.L().Info("heartbeat success with MasterNotReady")
-					}
```
Comment (amyangfei) [inline engine/executor/server.go:663]: This case is not checked after this PR.

Comment (gozssky) [reply engine/executor/server.go]: Addressed in https://github.com/pingcap/tiflow/pull/7362/commits/7dc5da1bf53b9fd03abf0373f8b120380f6b7d53.

**Comment by gozssky** [comment]
The integration test failed because the metastore could not be initialized.
```
[2022/10/27 06:27:58.276 +00:00] [ERROR] [server.go:515] ["create schema for framework metastore fail"] [schema=test_framework] [error="[DFLOW:ErrMetaOpFail]meta operation fail: context deadline exceeded"]
[2022/10/27 06:27:58.276 +00:00] [ERROR] [master.go:86] ["run dataflow server master with error"] [error="[DFLOW:ErrMetaOpFail]meta operation fail: context deadline exceeded"]
```

**Review by amyangfei — APPROVED** [review]

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**